client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts

```diff
@@ -0,0 +1,106 @@
+import { useSelector } from 'react-redux';
+
+import Desk from 'models/Desk';
+import axios from 'Utils/axios';
+import logger from 'logger/logger';
+import { Service, Severity } from 'models/Logger';
+import StoreStateType from 'redux/storeStateType';
+import InvestigatorOption from 'models/InvestigatorOption';
+import useCustomSwal from 'commons/CustomSwal/useCustomSwal';
+import InvestigationTableRow from 'models/InvestigationTableRow';
+
+import { InvestigationTableFooterOutcome, InvestigationTableFooterParameters } from './InvestigationTableFooterInterfaces';
+
+const useInvestigationTableFooter = (parameters: InvestigationTableFooterParameters): InvestigationTableFooterOutcome => {
+        
+    const { setOpenDesksDialog, setOpenInvestigatorsDialog, checkedRowsIds, tableRows, setTableRows } = parameters;
+
+    const { alertError } = useCustomSwal();
+
+    const userId = useSelector<StoreStateType, string>(state => state.user.id);
+
+    const handleOpenDesksDialog = () => setOpenDesksDialog(true);
+
+    const handleCloseDesksDialog = () => setOpenDesksDialog(false);
+
+    const handleConfirmDesksDialog = (updatedDesk: Desk, transferReason: string) => {
+        axios.post('/landingPage/changeDesk', {
+            epidemiologyNumbers: checkedRowsIds,
+            updatedDesk: updatedDesk.id,
+            transferReason
+        }).then(() => {
+            logger.info({
+                service: Service.CLIENT,
+                severity: Severity.LOW,
+                workflow: 'Switch desk',
+                step: 'the desk have been switched successfully',
+                investigation: checkedRowsIds.join(', '),
+                user: userId
+            })
+            const updatedRows : InvestigationTableRow[] = tableRows.map(row => 
```

Comment:
I see you use lines 40-43 in both functions (handleConfirmDesksDialog  and handleConfirmInvestigatorsDialog), maybe extract it to a function